OneSignal/OneSignal-Flutter-SDK
# Refinements to CI action

# Description

Refinements made to CI workflow to enable action to run to completion.

This PR also addresses most of the messages generated by the analyzer command.

```log
╭─iamwill@zues ~/code/@onesignal/OneSignal-Flutter-SDK ‹workflows-for-everyone› 
╰─$ flutter analyze --no-pub --no-fatal-infos                                                                                                                                   1 ↵
Analyzing OneSignal-Flutter-SDK...                                      

   info • Avoid using braces in interpolation when not needed • example/lib/main.dart:41:59 • unnecessary_brace_in_string_interps
   info • Avoid using braces in interpolation when not needed • example/lib/main.dart:50:51 • unnecessary_brace_in_string_interps
   info • The operand can't be null, so the condition is always false • example/lib/main.dart:130:16 • unnecessary_null_comparison
   info • The operand can't be null, so the condition is always false • example/lib/main.dart:268:21 • unnecessary_null_comparison
   info • The operand can't be null, so the condition is always false • example/lib/main.dart:278:21 • unnecessary_null_comparison
   info • 'FlatButton' is deprecated and shouldn't be used. Use TextButton instead. See the migration guide in flutter.dev/go/material-button-migration-guide). This feature was
          deprecated after v1.26.0-18.0.pre. • example/lib/main.dart:583:15 • deprecated_member_use
   info • 'FlatButton.FlatButton' is deprecated and shouldn't be used. Use TextButton instead. See the migration guide in flutter.dev/go/material-button-migration-guide). This
          feature was deprecated after v1.26.0-18.0.pre. • example/lib/main.dart:583:15 • deprecated_member_use
   info • The declaration '_processSettings' isn't referenced • lib/onesignal_flutter.dart:514:24 • unused_element
   info • The operand can't be null, so the condition is always false • lib/onesignal_flutter.dart:517:18 • unnecessary_null_comparison
   info • Unused import: 'package:flutter/foundation.dart' • lib/src/create_notification.dart:2:8 • unused_import
   info • The import of 'package:onesignal_flutter/src/notification.dart' is unnecessary because all of the used elements are also provided by the import of
          'package:onesignal_flutter/onesignal_flutter.dart' • lib/src/create_notification.dart:3:8 • unnecessary_import
   info • The import of 'package:onesignal_flutter/src/defines.dart' is unnecessary because all of the used elements are also provided by the import of
          'package:onesignal_flutter/onesignal_flutter.dart' • lib/src/notification.dart:1:8 • unnecessary_import
   info • Unused import: 'package:flutter/foundation.dart' • lib/src/notification.dart:4:8 • unused_import
   info • Use `;` instead of `{}` for empty constructor bodies • lib/src/outcome_event.dart:38:20 • empty_constructor_bodies
   info • The import of 'dart:async' is unnecessary because all of the used elements are also provided by the import of 'package:flutter_test/flutter_test.dart' •
          test/mock_channel.dart:1:8 • unnecessary_import
   info • Unused import: 'package:onesignal_flutter/src/defines.dart' • test/notification_test.dart:3:8 • unused_import
```

### Motivation

The primary step of this action is to run tests; however, info and warning messages from the analysis step were being treated as fatal.

# Testing

1. Navigate to the [CI workflow](https://github.com/OneSignal/OneSignal-Flutter-SDK/actions/workflows/ci.yml)
2. Click the _Run workflow_ dropdown button
3. Select the `workflows-for-everyone` branch
4. Click the _Run workflow_ primary button
5. Inspect the run to ensure things run to completion despite warning messages emitted by the analyze step

# Resources

###### Manually dispatch action
<img width="1274" alt="Screen Shot 2022-08-29 at 1 04 05 PM" src="https://user-images.githubusercontent.com/1715082/187288314-a7925bc7-81a0-40af-8d09-2f4fc58afe3b.png">

## References
* [GitHub Flutter Action](https://github.com/marketplace/actions/flutter-action)
* [Pubspec Author property is deprecated](https://dart.dev/tools/pub/pubspec#authorauthorsa)
* [Creating a verified publisher](https://dart.dev/tools/pub/publishing#create-verified-publisher)
* [Dart CLI Format command](https://dart.dev/tools/dart-format)


- - -
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/onesignal/onesignal-flutter-sdk/595)
